server/app/services/export/QuestionJsonSampler.java

```diff
@@ -0,0 +1,525 @@
+
+    private final QuestionJsonPresenter currencyJsonPresenter;
+
+    @Inject
+    CurrencyJsonSampler(QuestionJsonPresenter.Factory questionJsonPresenterFactory) {
+      this.currencyJsonPresenter = questionJsonPresenterFactory.create(QuestionType.CURRENCY);
+    }
+
+    @Override
+    public CurrencyQuestion getQuestion(ApplicantQuestion applicantQuestion) {
+      return applicantQuestion.createCurrencyQuestion();
+    }
+
+    @Override
+    public void addSampleData(ApplicantData applicantData, CurrencyQuestion question) {
+      applicantData.putCurrencyDollars(question.getCurrencyPath(), "123.45");
+    }
+
+    @Override
+    public QuestionJsonPresenter getJsonPresenter() {
+      return currencyJsonPresenter;
+    }
+  }
+
+  class DateJsonSampler implements QuestionJsonSampler<DateQuestion> {
+    private final QuestionJsonPresenter dateJsonPresenter;
+
+    @Inject
+    DateJsonSampler(QuestionJsonPresenter.Factory questionJsonPresenterFactory) {
+      this.dateJsonPresenter = questionJsonPresenterFactory.create(QuestionType.DATE);
+    }
+
+    @Override
+    public DateQuestion getQuestion(ApplicantQuestion applicantQuestion) {
+      return applicantQuestion.createDateQuestion();
+    }
+
+    @Override
+    public void addSampleData(ApplicantData applicantData, DateQuestion question) {
+      applicantData.putDate(question.getDatePath(), "2023-01-02");
+    }
+
+    @Override
+    public QuestionJsonPresenter getJsonPresenter() {
+      return dateJsonPresenter;
+    }
+  }
+
+  class EmailJsonSampler extends ContextualizedScalarsJsonSampler<EmailQuestion> {
+    private final QuestionJsonPresenter emailJsonPresenter;
+
+    @Inject
+    EmailJsonSampler(QuestionJsonPresenter.Factory questionJsonPresenterFactory) {
+      this.emailJsonPresenter = questionJsonPresenterFactory.create(QuestionType.EMAIL);
+    }
+
+    @Override
+    public EmailQuestion getQuestion(ApplicantQuestion applicantQuestion) {
+      return applicantQuestion.createEmailQuestion();
+    }
+
+    @Override
+    public void addSampleData(ApplicantData applicantData, EmailQuestion question) {
+      applicantData.putString(question.getEmailPath(), "homer.simpson@springfield.gov");
+    }
+
+    @Override
+    public QuestionJsonPresenter getJsonPresenter() {
+      return emailJsonPresenter;
+    }
+  }
+
+  class EmptyJsonSampler implements QuestionJsonSampler<Question> {
+
+    @Override
+    public CfJsonDocumentContext getSampleJson(QuestionDefinition questionDefinition) {
+      return new CfJsonDocumentContext();
+    }
+
+    @Override
+    public Question getQuestion(ApplicantQuestion applicantQuestion) {
+      return applicantQuestion.createCurrencyQuestion();
+    }
+
+    @Override
+    public void addSampleData(ApplicantData applicantData, Question question) {
+      // no-op
+    }
+
+    @Override
+    public QuestionJsonPresenter getJsonPresenter() {
+      return null;
```

Comment:
Should this return the EmptyJsonPresenter?

Reply:
There isn't an `EmptyJsonPresenter`, and no need for one. Since we override the default `getSampleJson` in `EmptyJsonSampler`, it always returns an empty `CfJsonDocumentContext` and we don't need to worry about it.